Syncing an empty database fails

If I create an empty database, and attempt to sync, it fails:

```
penberg@vonneumann libsql % cargo run --example offline_writes

warning: libsql-ffi@0.5.0: bundled/src/sqlite3.c:85489:9: warning: 'ALIGN' macro redefined [-Wmacro-redefined]
warning: libsql-ffi@0.5.0: #define ALIGN(n, size) (((n + size - 1) / size) * size)
warning: libsql-ffi@0.5.0:         ^
warning: libsql-ffi@0.5.0: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/arm/param.h:61:9: note: previous definition is here
warning: libsql-ffi@0.5.0: #define ALIGN(p)        __DARWIN_ALIGN(p)
warning: libsql-ffi@0.5.0:         ^
warning: libsql-ffi@0.5.0: bundled/src/sqlite3.c:211920:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
warning: libsql-ffi@0.5.0: }
warning: libsql-ffi@0.5.0: ^
warning: libsql-ffi@0.5.0: 2 warnings generated.
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.46s
     Running `/Users/penberg/src/tursodatabase/libsql/target/debug/examples/offline_writes`
Syncing database from remote...
thread 'main' panicked at libsql/examples/offline_writes.rs:43:21:
called `Result::unwrap()` on an `Err` value: Sync(PullFrame(500, "{\"status\": \"error\", \"generation\": 0, \"max_frame_no\": 0}"))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

--- a/libsql/src/sync.rs
+++ b/libsql/src/sync.rs
@@ -281,7 +281,9 @@ impl SyncContext {
                     .map_err(SyncError::HttpBody)?;
                 return Ok(PullResult::Frame(frame));
             }
-            if res.status() == StatusCode::BAD_REQUEST {
+            // BUG ALERT: The server returns a 500 error if the remote database is empty.
+            // This is a bug and should be fixed.
+            if res.status() == StatusCode::BAD_REQUEST || res.status() == StatusCode::INTERNAL_SERVER_ERROR {
                 let res_body = hyper::body::to_bytes(res.into_body())
                     .await
                     .map_err(SyncError::HttpBody)?;

